server/protocols/webadmin/webadmin-mailbox/src/main/java/org/apache/james/webadmin/service/EventDeadLettersRedeliverService.java

```diff
@@ -52,10 +52,8 @@ public EventDeadLettersRedeliverService(EventBus eventBus, EventDeadLetters dead
 
     private Mono<Task.Result> redeliverGroupEvents(Group group, Event event, EventDeadLetters.InsertionId insertionId) {
         return eventBus.reDeliver(group, event)
-            .then(Mono.fromCallable(() -> {
-                deadLetters.remove(group, insertionId);
-                return Task.Result.COMPLETED;
-            }))
+            .then(deadLetters.remove(group, insertionId)
+                .then(Mono.just(Task.Result.COMPLETED)))
```

Comment:
do you have a reason to nest the calls to `.then`?

Reply:
typing?